### PR TITLE
OCPBUGS-3816: oc whoami --show-console: check whether the Console capability is enabled

### PR DESCRIPTION
**Pull request by ingvagabund**
Perform additional check taking capabilities into account. If capabilities are not present, fall back to the previous behavior.

### CODE REVIEW

File: pkg/cli/whoami/whoami.go [hunk 1]
```diff
@@ -119,6 +129,28 @@ func (o *WhoAmIOptions) Validate() error {
...
+			if cap == configv1.ClusterVersionCapabilityConsole {
+				consoleEnabled = true
+				break
+			}
+		}
+	} else if errors.IsNotFound(err) {
+		// This means there's no way of telling whether the Console capability is enabled.
+		// Assume it is enabled unless proven otherwise.
+		consoleEnabled = true
+	} else {
```
Comment (ardaguclu) [inline pkg/cli/whoami/whoami.go:145]: nit: would be better to return early

**Review by ardaguclu — COMMENTED** [review]
/lgtm

File: pkg/cli/whoami/whoami.go [hunk 2]
```diff
@@ -119,6 +129,28 @@ func (o *WhoAmIOptions) Validate() error {
 }
 
 func (o *WhoAmIOptions) getWebConsoleUrl() (string, error) {
+	consoleEnabled := false
+	clusterVersion, err := o.ConfigClient.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
+	if err == nil {
+		for _, cap := range clusterVersion.Status.Capabilities.EnabledCapabilities {
```
Comment (ardaguclu) [inline pkg/cli/whoami/whoami.go:135]: It is good to learn how we can extract capabilities from cluster

**Comment by ardaguclu** [comment]
This seems genuine failure on disconnected https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_oc/1395/pull-ci-openshift-oc-master-e2e-metal-ipi-ovn-ipv6/1642773790649749504

**Comment by zhouying7780** [comment]
pre-merge tested , show the error info as expected

**Comment by ardaguclu** [comment]
Since failure is genuine;

/hold

**Comment by ingvagabund** [comment]
`[sig-cli] oc basics can show correct whoami result with console [Skipped:NoOptionalCapabilities] [Suite:openshift/conformance/parallel]` failing with:
```
msg: "Error running /usr/bin/oc --namespace=e2e-test-oc-basics-4c9j7 --kubeconfig=/tmp/configfile2492878406 whoami --show-console:\nStdOut>\nerror: unable to determine console location: clusterversions.config.openshift.io \"version\" is forbidden: User \"e2e-test-oc-basics-4c9j7-user\" cannot get resource \"clusterversions\" in API group \"config.openshift.io\" at the cluster scope\nStdErr>\nerror: unable to determine console location: clusterversions.config.openshift.io \"version\" is forbidden: User \"e2e-test-oc-basics-4c9j7-user\" cannot get resource \"clusterversions\" in API group \"config.openshift.io\" at the cluster scope\nexit status 1\n",
```

**Comment by ingvagabund** [comment]
Unfortunately, the clusterversion config object can be read only by admins.

**Pull request closed without merging**